Unable to build message packages on top of nightly archives

## Bug report

**Required Info:**

- Operating System:
  - Ubuntu Focal
- Installation type:
  - nightly archive
- Version or commit hash:
  - works with archive from https://ci.ros2.org/view/packaging/job/packaging_linux/1811/
  - breaks with archive from https://ci.ros2.org/view/packaging/job/packaging_linux/1819/
- DDS implementation:
  - Connext *not* installed
- Client library (if applicable):
  - <!-- e.g. rclcpp, rclpy, or N/A -->

#### Steps to reproduce issue
<!-- Detailed instructions on how to reliably reproduce this issue http://sscce.org/
``` code that can be copy-pasted is preferred ``` -->
```
docker pull osrf/ros2:nightly
apt update && apt install -y libssl-dev
cd && mkdir -p ws/src && cd ws
git clone https://github.com/ros2/example_interfaces src/example_interfaces
colcon build
```

#### Expected behavior

```
Starting >>> example_interfaces
Finished <<< example_interfaces [24.2s]                       

Summary: 1 package finished [24.3s]
```

#### Actual behavior

```
Starting >>> example_interfaces
--- stderr: example_interfaces                         
CMake Error at /opt/ros/foxy/share/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake:276 (add_dependencies):
  The dependency target "example_interfaces__rosidl_typesupport_connext_c" of
  target "example_interfaces__rosidl_typesupport_connext_c__pyext" does not
  exist.
Call Stack (most recent call first):
  /opt/ros/foxy/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/foxy/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:286 (ament_execute_extensions)
  CMakeLists.txt:16 (rosidl_generate_interfaces)


CMake Generate step failed.  Build files cannot be regenerated correctly.
---
Failed   <<< example_interfaces	[ Exited with code 1 ]

```

#### Additional information

It works with the nightly archive from  April 3rd: https://ci.ros2.org/view/packaging/job/packaging_linux/1811/
Fails with the next successfully built archive from April 11th: https://ci.ros2.org/view/packaging/job/packaging_linux/1819/


It I install connext and setup the environment it works:
```
docker pull osrf/ros2:nightly-rmw-nonfree
apt update && apt install -y libssl-dev
source /opt/rti.com/rti_connext_dds-5.3.1/resource/scripts/rtisetenv_x64Linux3gcc5.4.0.bash 
cd && mkdir -p ws/src && cd ws
git clone https://github.com/ros2/example_interfaces src/example_interfaces
colcon build
```

Issue also reported at https://answers.ros.org/question/349612/ros2-osrf-docker-container-fails-to-build/

--- a/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake
+++ b/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake
@@ -221,6 +221,11 @@ rosidl_target_interfaces(${_target_name_lib}
 
 foreach(_typesupport_impl ${_typesupport_impls})
   find_package(${_typesupport_impl} REQUIRED)
+  # a typesupport package might not be able to generated a target anymore
+  # (e.g. if an underlying vendor package isn't available in an overlay)
+  if(NOT TARGET ${rosidl_generate_interfaces_TARGET}__${_typesupport_impl})
+    continue()
+  endif()
 
   set(_pyext_suffix "__pyext")
   set(_target_name "${PROJECT_NAME}__${_typesupport_impl}${_pyext_suffix}")

